src/NuGet.Core/NuGet.Commands/RestoreCommand/Diagnostics/UnresolvedMessages.cs

```diff
@@ -264,9 +264,11 @@ internal static NuGetVersion GetBestMatch(SortedSet<NuGetVersion> versions, Vers
                 }
             }
 
-            if (!range.IsFloating || range.HasLowerBound || range.HasUpperBound)
+            // Take the lowest version higher than the pivot if one exists,
+            // unless the version range is floating without an upper bound
```

Comment:
Am I reading this wrong? The comment says "unless" (aka, if there's NOT) an upper bound. But the code says "floating AND hasUpperBound".

Reply:
I think the comment is wrong. 
Basically if there's an upper bound we take the lowest version above the bound. 

So I think I'll change it to: 

"Take the lowest version higher than the pivot if one exists or for ranges that are floating with an upper bound"

Any suggestions? 

The longer explanation is: 

* Floatless ranges - If there's a pivot/lower or higher bound, take the first above that pivot.
* Floating ranges - by definition they need the latest version, so unless there's an upper bound always show the highest version possible.

Reply:
Sure, I can do that.

Reply:
Having a truth table with common cases as rows and columns of these observed properties could help the reader (code comment)

Reply:
Aren't the tests the truth table? :) 
Feels like I'd be repeating the tests.

Reply:
The tests have comments that go in that much detail.

Reply:
I've found that big code comments in the product code -- more of the "why" than the "how" -- help explain code that could look weird to a new reader. I feel like the reality is that test code does not get the same kind of understanding/process/quality checks as product code. In particular if someone is confused about some behavior it's pretty easy to find the product code on GitHub but less easy to find the related tests to explain the behavior. Just a suggestion though -- your call.

Reply:
I'll look into it. 

I just don't want to be too verbose ;)

Reply:
> @joelverhagen Added a truth table, ready for another look.
> 
> @donnie-msft Let me if you think other examples are needed.

Looks good. I'm not a fan of the truth table in the code since you are already showing examples in the tests. But I understand why some would prefer to see it here.